Update MLE fit parameter initialisation

aherbert opened the pull request:
Updated the initialisation of the start parameters to use the values from the input configuration. The result is the same as the hardcoded "10^(-1 + 2*rand())" but can be changed via configuration.

The initialisation of the 2nd and 3rd species for konstart must be expressed relative to koffstart (as is done for species 1) and not absolute (i.e. multiplied by koffstart).

This change also adds the best nll score to the returned values as this is of use when selecting fits with different settings, e.g. number of species or fixed parameters.

aherbert committed:
Update MLE fit parameter initialisation

Updated the initialisation of the start parameters to use the values from the input configuration. The result is the same as the hardcoded "10^(-1 + 2*rand())" but can be changed via configuration.

The initialisation of the 2nd and 3rd species for konstart must be expressed relative to koffstart (as is done for species 1) and not absolute (i.e. multiplied by koffstart).

This change also adds the best nll score to the returned values as this is of use when selecting fits with different settings, e.g. number of species or fixed parameters.

JochemVink commented:
Thanks a lot for your efforts. Sorry this was not correctly implemented in the original version. How is it going with analyzing your data?

aherbert commented:
Hi. I've only found a few issues with the code that are minor and do not really effect the output. I did notice in the function `SimulationLocalizationandConfinement.m` that you use an approximation to the CDF of the exponential function to set the state switching probability:
```
kt = 1 - e^-kt
```
This is valid when kt is small. Using the default Steptime t of 1e-5 you need to have a rate constant k of about 2000 to see a relative error between the two of 1%:
```
1e-5 * 2000 ~ 1 - e^(-1e-5 * 2000)
0.02 ~ 0.0198
```

The approximation kt is higher and so there will be more state switches. At lower k the relative error is much lower. In real terms with the rate constants I am using this has no effect but I updated my version as the change has no impact on run time. Would you like a PR to show the change?

I have managed to simulate a 3 state, 1 species model using the example `Figure4BC.m` as a start point. I updated the simulation using various settings that are a guess for the diffusion and binding rates for our protein. In all cases I tried anaDDA is able to fit the data it simulates. However it does not come close to fitting the real track data we have for our protein. So there must be some extra kinetics occurring that we do not understand. Currently we are collecting some more data at a different frame rate so that we can use that functionality of anaDDA to increase the information for the fit.

JochemVink commented:
Hi Alex,

yes your modification of the simulation script sounds like another good addition, so would be happy with a PR there.

A word about using different frame times. So in general, that trick works best for when you have a good fit to your data already and several other models that you think could be occurring also fit well. You then use data from different frame times to differentiate between these models.

In your case you say that the model does not fit well with the data. So in that case that will probably only get worse if you add in further data that it will also need to fit. Maybe you can send an example of what the fit looks like and I can have a look and see what might be a reason for it. Is it a system composed of subunits within a larger complex? Is it a eukaryotic/prokaryotic system?  Is it measured with stroboscopic illumination or continuous exposure?

Cheers,

Jochem
________________________________
From: Alex Herbert ***@***.***>
Sent: Wednesday, March 10, 2021 15:45
To: HohlbeinLab/anaDDA ***@***.***>
Cc: JochemVink ***@***.***>; State change ***@***.***>
Subject: Re: [HohlbeinLab/anaDDA] Update MLE fit parameter initialisation (#4)


Hi. I've only found a few issues with the code that are minor and do not really effect the output. I did notice in the function SimulationLocalizationandConfinement.m that you use an approximation to the CDF of the exponential function to set the state switching probability:

kt = 1 - e^-kt


This is valid when kt is small. Using the default Steptime t of 1e-5 you need to have a rate constant k of about 2000 to see a relative error between the two of 1%:

1e-5 * 2000 ~ 1 - e^(-1e-5 * 2000)
0.02 ~ 0.0198


The approximation kt is higher and so there will be more state switches. At lower k the relative error is much lower. In real terms with the rate constants I am using this has no effect but I updated my version as the change has no impact on run time. Would you like a PR to show the change?

I have managed to simulate a 3 state, 1 species model using the example Figure4BC.m as a start point. I updated the simulation using various settings that are a guess for the diffusion and binding rates for our protein. In all cases I tried anaDDA is able to fit the data it simulates. However it does not come close to fitting the real track data we have for our protein. So there must be some extra kinetics occurring that we do not understand. Currently we are collecting some more data at a different frame rate so that we can use that functionality of anaDDA to increase the information for the fit.

—
You are receiving this because you modified the open/close state.
Reply to this email directly, view it on GitHub<https://github.com/HohlbeinLab/anaDDA/pull/4#issuecomment-795527037>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AHAX6Y2GMLWKOISK4XWCFSDTC6AX7ANCNFSM4YRDQQYA>.
